src/plugins/discover/public/application/main/utils/resolve_data_view.ts

```diff
@@ -141,21 +145,45 @@ export function resolveDataView(
       });
       return ownDataView;
     }
-    if (!Boolean(isTextBasedQuery)) {
```

Comment:
No, it wasn't but I guess it was a change that got lost since it was added during the long period of time of this PR ... These message doesn't make sense when loading a new saved search with SQL in the state URL. I reverted this change to consider text based query , also added a test for it: https://github.com/elastic/kibana/pull/153528/commits/53ec89dc6b0d92d1916e466ca78d309b68a35d78